Sources/_StringProcessing/PrintAsPattern.swift

```diff
@@ -180,11 +190,118 @@ extension PrettyPrinter {
 
   // TODO: Some way to integrate this with conversion...
   mutating func printAsPattern(
-    _ ccc: DSLTree.CustomCharacterClass
+    _ ccc: DSLTree.CustomCharacterClass,
+    terminateLine: Bool = true
   ) {
-    let inv = ccc.isInverted ? "inverted: true" : ""
-    printBlock("CharacterClass(\(inv))") { printer in
-      ccc.members.forEach { printer.printAsPattern($0) }
+    if ccc.hasUnprintableProperty {
+      printAsRegex(ccc, terminateLine: terminateLine)
+      return
+    }
+    
+    defer {
+      if ccc.isInverted {
+        printIndented { printer in
+          printer.indent()
+          
+          if terminateLine {
+            printer.print(".inverted")
+          } else {
+            printer.output(".inverted")
+          }
+        }
+      }
+    }
+    
+    // If we only have 1 member, then we can emit it without the extra
+    // CharacterClass initialization
+    if ccc.members.count == 1 {
+      printAsPattern(ccc.members[0])
+      
+      if terminateLine {
+        self.terminateLine()
+      }
+      
+      return
+    }
+    
+    var charMembers = ""
+    
+    let nonCharMembers = ccc.members.filter {
```

Comment:
What is this doing or for? Add a comment, and consider if it's worth giving the concept a name e.g. a computed var